lib/typing_env.ml

```diff
@@ -45,19 +45,27 @@ let replace_matching_ids ~reference ~current =
                   Sig_type (new_id, td, r, v) :: lst )
             | None -> (subst, sig_typ_decl :: lst))
         | Sig_module (id, mp, md, r, v) as sig_mod_decl -> (
-            match Env.find_value_index id ref_env with
+            match Env.find_module_index id ref_env with
             | Some _ ->
                 let new_id = Ident.rename id in
                 ( Subst.add_module id (Path.Pident new_id) subst,
                   Sig_module (new_id, mp, md, r, v) :: lst )
             | None -> (subst, sig_mod_decl :: lst))
-        | Sig_modtype (id, mtd, v) as sig_modtyp_decl -> (
+        | Sig_modtype (id, mtd, v) -> (
             match Env.find_modtype_index id ref_env with
             | Some _ ->
                 let new_id = Ident.rename id in
                 ( Subst.add_modtype id (Mty_ident (Pident new_id)) subst,
                   Sig_modtype (new_id, mtd, v) :: lst )
-            | None -> (subst, sig_modtyp_decl :: lst))
+            | None ->
+              (** This is a special case for functor paramters, should be
```

Comment:
Could you provide a bit more explanation here? I think you found a pretty nasty bug and it'd be great to have it documented in the history somewhere at least!

We might even add that to the documentation of the whole function as a justification to why rewriting IDs is necessary as long as we rely on the compiler's functions.

Reply:
I added some documentation, feel free to change it if it's not clear enough.